app/src/main/java/com/yalta/repositories/RealRepo.kt

```diff
@@ -3,31 +3,47 @@ package com.yalta.repositories
 import com.yalta.services.SessionService
 import okhttp3.*
 import ru.gildor.coroutines.okhttp.await
+import java.net.ConnectException
+import java.net.SocketTimeoutException
 
 open class RealRepo {
     private val baseUrl = "http://10.0.2.2:9000"
     private val client = OkHttpClient()
 
-    suspend fun doPostRequest(url : String, body : String) : Response {
+    suspend fun doPostRequest(url : String, body : String) : Response? {
```

Comment:
Yeah, I don't really like the idea of introducing Nullable in new Kotlin-only API...

Maybe at least use `Either`

Reply:
Done